app/src/androidTest/java/com/arygm/quickfix/ui/profile/UpgradetoWorkerScreenTest.kt

```diff
@@ -0,0 +1,259 @@
+    composeTestRule
+        .onNodeWithTag("locationInput")
+        .assert(SemanticsMatcher.expectValue(SemanticsProperties.EditableText, AnnotatedString("")))
+
+    // Cliquer sur le bouton de validation
+    composeTestRule.onNodeWithTag("validateButton").performClick()
+
+    // Vérifier que le message d'erreur est affiché
+    composeTestRule.onNodeWithTag("errorMessage").assertTextEquals("Please fill all fields")
+  }
+
+  @Test
+  fun testOccupationDropdownMenuDisplaysOptions() {
+    composeTestRule.setContent { QuickFixTheme { BusinessScreen(navigationActions) } }
+
+    // Cliquer sur le champ "Occupation" pour ouvrir le menu déroulant
+    composeTestRule.onNodeWithTag("occupationDropdownIcon").performClick()
+
+    // Vérifier que les options du menu déroulant sont affichées
+    val occupations = listOf("Carpenter", "Painter", "Plumber", "Electrician", "Mechanic")
+    occupations.forEach { occupation ->
+      composeTestRule.onNodeWithText(occupation).assertIsDisplayed()
+    }
+  }
+
+  @Test
+  fun testSelectingOccupationUpdatesField() {
+    composeTestRule.setContent { QuickFixTheme { BusinessScreen(navigationActions) } }
+
+    // Ouvrir le menu déroulant
+    composeTestRule.onNodeWithTag("occupationDropdownIcon").performClick()
+
+    // Sélectionner "Plumber"
+    composeTestRule.onNodeWithText("Plumber").performClick()
+
+    // Vérifier que le champ "Occupation" est mis à jour
+    composeTestRule.onNodeWithTag("occupationInput").assertTextEquals("Plumber")
+  }
+
+  @Test
+  fun testHourlyRateAcceptsOnlyDigits() {
+    composeTestRule.setContent { QuickFixTheme { BusinessScreen(navigationActions) } }
+
+    // Entrer une valeur avec des lettres et des chiffres
+    composeTestRule.onNodeWithTag("hourlyRateInput").performTextInput("abc123def")
+
+    // Vérifier que seuls les chiffres sont acceptés
+    composeTestRule.onNodeWithTag("hourlyRateInput").assertTextContains("123")
+    composeTestRule.onNodeWithTag("hourlyRateInput").performTextClearance()
+
+    composeTestRule.onNodeWithTag("hourlyRateInput").performTextInput("123")
+
+    // Vérifier que seuls les chiffres sont acceptés
+    composeTestRule.onNodeWithTag("hourlyRateInput").assertTextContains("123")
+  }
+}
+
+class FakeProfileRepository : ProfileRepository {
+  internal val profiles = mutableListOf<Profile>()
+  private val loggedInProfileFlow = MutableStateFlow<Profile?>(null)
+
+  override fun init(onSuccess: () -> Unit) {
+    onSuccess()
+  }
+
+  override fun getProfiles(onSuccess: (List<Profile>) -> Unit, onFailure: (Exception) -> Unit) {
+    onSuccess(profiles)
+  }
+
+  override fun filterWorkers(
+      hourlyRateThreshold: Double?,
+      fieldOfWork: String?,
+      onSuccess: (List<Profile>) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    TODO("Not yet implemented")
+  }
+
+  override fun addProfile(profile: Profile, onSuccess: () -> Unit, onFailure: (Exception) -> Unit) {
+    TODO("Not yet implemented")
+  }
+
```

Comment:
We should use Mockito instead of manually mocking, and we can use 'whenever' or 'when' for the desired outcome.

Reply:
Good point! Using Mockito with whenever or when is indeed a cleaner and more standardized approach for mocking. I'll refactor the code to replace manual mocks with Mockito mocks and use whenever for setting up the desired outcomes. Thanks for the suggestion